File: Python/백준/Bronze/21180. Reconstruct Sum/Reconstruct Sum.py
n = int(input())
lst = [int(input()) for _ in range(n)]
sum_val = sum(lst)
ans = 100001
for i in range(n):
    if lst[i] == sum_val - lst[i]:
        ans = lst[i]
        break
print("BAD" if ans >= 100001 else ans)

File: C++17/백준/Bronze/21180. Reconstruct Sum/Reconstruct Sum.cc
#include <iostream>
#include <vector>

using namespace std;

int main() {
	int n;
	int sum = 0;
	int ans = 100001;
	cin >> n;
	vector<int> lst(n);
	for (int i = 0; i < n; i++) {
		cin >> lst[i];
		sum += lst[i];
	}
	for (int i = 0; i < n; i++) {
		if (lst[i] == sum - lst[i]) {
			ans = lst[i];
			break;
		}
	}
	if (ans < 100001) {
		cout << ans;
	}
	else {
		cout << "BAD";
	}
}

